lib/githubbot.ts

```diff
@@ -251,7 +252,7 @@ export class GithubBot extends ProcBot.ProcBot<string> {
                         // If there's more retries, try again in 5 seconds.
                         setTimeout(runApi, 5000);
                     }
-                }).then(resolve);
+                });
             };
```

Comment:
Oooh, in fact that refactor has just shown me a bug: if you have `retriesLeft === 1` at the start of `runApi`/the while loop, and you then get your first authentication error, then you skip the `retriesLeft === 0` check, end up with `retriesLeft === -1`, and this function retries forever without terminating.

Either way, we should definitely fix that (probably by checking `retries === 0` before the auth failure test, not after)